### ISSUE TITLE
Add S3 tile caching setup to terraform config

### ISSUE THREAD

**Issue by KlaasH** [azavea/pfb-network-connectivity#634]
See the [issue comment](https://github.com/azavea/tilegarden/issues/126#issuecomment-441776311) and [branch](https://github.com/azavea/tilegarden/tree/feature/kjh/s3-tile-cache).  The idea is to set things up so that
- tile requests come into the CloudFront distribution (as they do already)
- CloudFront sends them to an S3 Website set up to serve a bucket of cached tiles
- The S3 website configuration is set to redirect back to the CloudFront distribution, prepending `latest/` to the request path
- The CloudFront distribution sends all requests that start with `latest/` to the API gateway

The first request for a tile will thus be filled by API Gateway, i.e. by an actual Lambda function spinning up and making the tile. But before returning it, the Lambda function will write the tile to the bucket so later requests for the same tile will find it there.  The code changes are in a separate issue: #___.

So the task is to set up the Terraform config to provision CloudFront and S3.  Until the code side is done, all requests will fall back to the API gateway (though we could manually add tiles to the bucket to test the cache response).

**Comment by jeancochrane** [comment]
Currently blocked waiting on #680.
